scripts/coordinate.js

```diff
@@ -213,125 +213,121 @@ FriendlyChat.prototype.removeChats = function() {
-      var renderCreatorIcon = '';
-      var renderDoerIcons = '';
-      var appUsersRef = database.ref('app_users');
-      var doerEmails = whoInformation.split(', ');
-      var creatorEmail = data.val().creatorDescription;
-
-      appUsersRef.once('value')
-      .then(function(snap) {
-        doerUserObjects.length = 0; // Resetting the global variable.
-        creatorUserObjects.length = 0;
-
-        snap.forEach(function(childSnap) {
-          doerEmails.filter(function(doerEmail) {
-            if (doerEmail === childSnap.val().email) {
-              // console.log('it matches!');
-              // console.log(childSnap.val());
-              // console.log('childsnap.val', childSnap.val());
-              doerUserObjects.push(childSnap.val());
-            }
-          });
-          if (creatorEmail === childSnap.val().email) {
-            creatorUserObjects.push(childSnap.val());
+function renderDoWhopMainHeader(userSessionCurrentDoWhop) {
+  var currentDoWhopID = userSessionCurrentDoWhop; // This is available from higher scope.
+  firebase.database().ref().child('DoWhopDescriptions/' + userSessionCurrentDoWhop).once('value').then(function(data) {
+    var doWhopSelector = document.getElementById('dowhop-selector-container');
+    var doWhopSelectorDiv = '';
+    var imageUrl =
+      (data.val().downloadURL && data.val().downloadURL.image1) ||
+      data.val().downloadURL ||
+      defaultDoWhopDescriptionImage;
+
+    var doWhopDescriptionTitle = data.val().titleDescription || 'Your DoWhops Will Appear Here';
+
+    doWhopSelectorDiv +=
+      "<section id='" +
+      data.key +
+      "' class='dowhop-selector-block''>" +
+      "<div class='dowhop-selector-header-top' style='background-image: url(" +
+      imageUrl +
+      ");'>" +
+      '<h1>' +
+      doWhopDescriptionTitle +
+      '</h1>' +
+      '</div>';
+
+    doWhopSelector.innerHTML = doWhopSelectorDiv; // Compose header block only.
+  });
+}
+
+function generateUserIcons(userObjectArray) {
+  var textOutput = '';
+  if (userObjectArray && userObjectArray.length > 0) {
+    _.map(userObjectArray, function(userObject) {
+      textOutput +=
+        '<div class="user-avatar-container">' +
+        "<a href='/profile.html?" +
+        userObject.uid +
+        "'>" +
+        "<div class='user-avatar'>" +
+        "<img class='user-avatar' src='" +
+        userObject.photoURL +
+        "'>" +
+        '</img>' +
+        '</div>' +
+        '<div class="user-handle">' +
+        userObject.displayName +
+        '</div>' +
+        '</a>' +
+        '</div>';
+    });
+  }
+  return textOutput;
+}
+
+function checkDoWhopDetails(userSessionCurrentDoWhop) {
+  var currentDoWhopID = userSessionCurrentDoWhop;
+  // Turning on a listener to the DoWhopDescriptions for a given DoWhop ID:
+  firebase.database().ref().child('DoWhopDescriptions/' + userSessionCurrentDoWhop).on('value', function(data) {
```

Comment:
Could be more readable as:
database.ref('DoWhopDescriptions).child(currentDoWhopID)
  .on('value', function(data) {});